plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsRepository.java

```diff
@@ -107,43 +115,105 @@ protected void doStart() {
     }
```

Comment:
Commenting here since the line that I want to comment on is not changed here. I see that `doStart` catches an `IOException` and wraps it in an `ElasticsearchGenerationException`. That seems odd to me, I think this should be either `UncheckedIOException` or `RuntimeException`.